### PR TITLE
ci: Reorganize CI

### PR DESCRIPTION
**Pull request by LecrisUT**
- Rename the CI jobs so that they are more legible in the report
- Fix concurrency on the imported jobs
- Separated pre-commit to its own step
- ~~Add a static-analysis checker. I was experimenting with `sonarcloud` at [`CMake-Template`](https://github.com/LecrisUT/CMake-Template) repo. I think it could be useful to include it as an intermediate to `clang-tidy`. It is also polyglot, so it might pick up on more nieche problems~~. Postponed to #400
- Mask overall CI status on main branch
- Badges!!!

Depends on: #361 #358

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,21 @@
-[![Version Badge](https://anaconda.org/conda-forge/spglib/badges/version.svg)](https://anaconda.org/conda-forge/spglib)
-[![Downloads Badge](https://anaconda.org/conda-forge/spglib/badges/downloads.svg)](https://anaconda.org/conda-forge/spglib)
-[![PyPI](https://img.shields.io/pypi/dm/spglib.svg?maxAge=2592000)](https://pypi.python.org/pypi/spglib)
-
 # Spglib
 
+[![CI Status][ci-badge]][ci-link]
+[![Documentation Status][rtd-badge]][rtd-link]
+[![Codecov Status][codecov-badge]][codecov-link]
```
Comment (LecrisUT) [inline README.md:5]: Codecov badge needs a token

File: README.md [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+
+![License Status][license-badge]
+![CMake Status][cmake-badge]
+[![Python Versions][python-version]][pypi-link]
```
Comment (LecrisUT) [inline README.md:9]: Python version badge will be fixed automatically in next pypi release

File: README.md [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+
+[![Release Status][release-badge]][release-link]
+[![Conda Status][conda-badge]][conda-link]
+[![PyPI Status][pypi-badge]][pypi-link]
+[![Fedora Status][fedora-badge]][fedora-link]
+
+[![PyPI Downloads][pypi-downloads]][pypi-link]
+[![Conda Downloads][conda-downloads]][conda-link]
```
Comment (LecrisUT) [inline README.md:17]: Most badges are from https://shields.io/badges. There could be more interesting ones to find. I think a social one would be good, but there is no matrix/discord room etc. Doesn't have to be spglib specific, just a place where developers can be contacted in real-time

**Comment by lan496** [comment]
@LecrisUT I've received "Request to install Codecov in spglib". Is it needed for the codecov badge? I remember this project is already integrated with codecov. What is the difference?

**Comment by LecrisUT** [comment]
> @LecrisUT I've received "Request to install Codecov in spglib". Is it needed for the codecov badge? I remember this project is already integrated with codecov. What is the difference?

Can you get the badge api-token without installing codecov app? That's the main thing I'm trying with that.

**Comment by lan496** [comment]
Do you need the badge api token as GitHub's Action secrets or raw string?

**Comment by LecrisUT** [comment]
> Do you need the badge api token as GitHub's Action secrets or raw string?

Raw string, it's meant to be public and used in the README.md

**Comment by lan496** [comment]
I see. Then, here it is.
```
[![codecov](https://codecov.io/gh/spglib/spglib/graph/badge.svg?token=G91xAWu1JT)](https://codecov.io/gh/spglib/spglib)
```

**Comment by LecrisUT** [comment]
All badges except for Python version are working. Just 2 things for this PR:
- Feedback on namings:
  - I am considering `CI / 📘 docs / Sphinx-html` instead, but would that be confusing?
  - Would `🧪` be sufficient to indicate experimental jobs?
  - `+` sign is excessive?
  - Other suggestions for emojis?
- When ready to merge, the `pass` job changed name a bit.

File: README.md (hunk 3 above)
Comment (lan496) [reply README.md:17]: I think the GitHub issue and discussion work as such a community room now. So, I do not feel the need for another social one so much.

**Comment by lan496** [comment]
> Mask overall CI status on main branch

Can you explain this a little more? Are these badges reflected only on main branch?

File: README.md [hunk 4]
```diff
@@ -177,3 +189,23 @@ improvements. See [Contributing.md](Contributing.md) for more details.
 See [https://spglib.readthedocs.io/](https://spglib.readthedocs.io/) for further
 documentation. See the [doc documentation](doc/README.md) for more information on
 how to contribute to the documentation.
+
+[ci-badge]: https://github.com/LecrisUT/CMake-Template/actions/workflows/ci.yaml/badge.svg?branch=main&event=push
```
Comment (lan496) [inline README.md]: Is this link correct? It seems to link to your action.

**Comment by lan496** [comment]
> Feedback on namings:

I think the current emojis are nice!

**Comment by LecrisUT** [comment]
> > Mask overall CI status on main branch
> 
> Can you explain this a little more? Are these badges reflected only on main branch?

It's easiest with an example:
If you open the CI checks for [this PR](https://github.com/LecrisUT/CMake-Template/pull/10), you will see `windows` is failing, but the [merge commit](https://github.com/LecrisUT/CMake-Template/commit/a4f73dbd53d8cb98e6325296c2905b2ed40dcaa3) is showing it as passing. That is because I am masking the true result there so that on the [main page](https://github.com/LecrisUT/CMake-Template) the first commit is shown with a check mark.

File: .github/workflows/step_test.yaml [hunk 5]
```diff
@@ -1,52 +1,50 @@
...
     container: ${{ !matrix.os && 'ghcr.io/lecrisut/dev-env:main' || '' }}
-    needs: [ pre-commit ]
     continue-on-error: ${{ matrix.experimental || false }}
     strategy:
       fail-fast: false
       matrix:
-        toolchain: [ gcc, llvm, intel, windows, macos ]
-        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12"]
+        toolchain: [ gcc, llvm, intel ]
+        python-version: [ "3.8", "3.x" ]
```
Comment (lan496) [inline .github/workflows/step_test.yaml:29]: Note: "3.x" specifies the latest stable version.
https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input

**Comment by lan496** [comment]
I see. Thanks, I've left a few comments.

File: README.md (hunk 4 above)
Comment (LecrisUT) [reply README.md]: Oops, thanks for the catch :D

**Comment by lan496** [comment]
> Add a static-analysis checker. I was experimenting with sonarcloud at CMake-Template repo. I think it could be useful to include it as an intermediate to clang-tidy. It is also polyglot, so it might pick up on more nieche problems

Is it included?

**Comment by LecrisUT** [comment]
> > Add a static-analysis checker. I was experimenting with sonarcloud at CMake-Template repo. I think it could be useful to include it as an intermediate to clang-tidy. It is also polyglot, so it might pick up on more nieche problems
> 
> Is it included?

Not yet, I will postpone it for after release #400

File: README.md [hunk 6]
```diff
@@ -177,3 +189,23 @@ improvements. See [Contributing.md](Contributing.md) for more details.
 See [https://spglib.readthedocs.io/](https://spglib.readthedocs.io/) for further
 documentation. See the [doc documentation](doc/README.md) for more information on
 how to contribute to the documentation.
+
+[ci-badge]: https://github.com/spglib/spglib/actions/workflows/ci.yaml/badge.svg?branch=develop&event=push
+[ci-link]: https://github.com/spglib/spglib/actions/workflows/ci.yaml?query=branch%3Adevelop+event%3Apush
+[cmake-badge]: https://img.shields.io/badge/CMake-3.15-blue?logo=data:image/svg%2bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGRhdGEtbmFtZT0iTGF5ZXIgMiIgdmlld0JveD0iMCAwIDU0NS41IDU0NS45Ij48cGF0aCBkPSJNNTQ2IDUzNCAyODIgOWwzOSA0MzQgMjI1IDkxeiIgZGF0YS1uYW1lPSJyZWQgcmlnaHQiIHN0eWxlPSJmaWxsOiNmMzI3MzU7c3Ryb2tlLXdpZHRoOjAiLz48cGF0aCBkPSJNNTI1IDU0NiAxNzAgNDAzIDEgNTQ2aDUyNHoiIGRhdGEtbmFtZT0iZ3JlZW4gYm90dG9tIiBzdHlsZT0ic3Ryb2tlLXdpZHRoOjA7ZmlsbDojM2VhZTJiIi8+PHBhdGggZD0iTTI2MyAwIDAgNTIybDI4OC0yNDRMMjYzIDB6IiBkYXRhLW5hbWU9ImJsdWUgbGVmdCIgc3R5bGU9ImZpbGw6IzAwNjhjNztzdHJva2Utd2lkdGg6MCIvPjxwYXRoIGQ9Im0yOTEgMjk5LTEwNSA4OSAxMTcgNDgtMTItMTM3eiIgZGF0YS1uYW1lPSJncmF5IGNlbnRlciIgc3R5bGU9ImZpbGw6I2RjZTNlYztzdHJva2Utd2lkdGg6MCIvPjwvc3ZnPg==
```
Comment (lan496) [inline README.md:195]: Note: We have to manually update the badge for the CMake minimum version.

**Review by lan496 — APPROVED** [review]

**Review by atztogo — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (LecrisUT) [reply README.md:195]: Yes, specifically only the `3.15` text and only when we bump minimum version

**Comment by lan496** [comment]
@LecrisUT Can you address the branch conflicts?

**Comment by LecrisUT** [comment]
> @LecrisUT Can you address the branch conflicts?

@lan496 Done. I forgot, you need to switch the required CI for the auto-merge

**Comment by lan496** [comment]
Current branch rules
image

**Pull request merged**